### PR TITLE
:seedling: Remove ginkgo from internal/controller/validators

### LINKED ISSUE #193: Update controllers/validators package tests
## Description
---
Follow the testing guidelines set forth in #188 and refactor the [controllers/validators package](https://github.com/operator-framework/operator-controller/tree/main/internal/controllers/validators) unit tests.

### Pre-Conditions
---
- [ ] #188 

### Acceptance Criteria
---
- [ ] All test files strictly under the resolution package updated to conform to the guidelines in #188

### PR DESCRIPTION
**Pull request by tmshort**
Fix #193

Added additional semver tests and converted to a table.

# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: internal/controllers/validators/validators_test.go [hunk 1]
```diff
@@ -1,63 +1,94 @@
...
-		It("should not return an error for empty SemVer", func() {
-			operator := &v1alpha1.Operator{
-				Spec: v1alpha1.OperatorSpec{
-					Version: "",
-				},
-			}
-			err := validators.ValidateOperatorSpec(operator)
-			Expect(err).NotTo(HaveOccurred())
-		})
+type testData struct {
```
Comment (stevekuznetsov) [inline internal/controllers/validators/validators_test.go:13]: Scoping comments for this struct and data apply here (why shouldn't another test for another behavior in this package have their own `testData`?)

File: internal/controllers/validators/validators_test.go [hunk 2]
```diff
@@ -1,63 +1,94 @@
...
-				},
-			}
-			err := validators.ValidateOperatorSpec(operator)
-			Expect(err).NotTo(HaveOccurred())
-		})
-	})
-})
+func TestValidateOperatorSpecSemVer(t *testing.T) {
+	for _, d := range sm {
+		t.Logf("Testing %s: %q", d.comment, d.data)
```
Comment (stevekuznetsov) [inline internal/controllers/validators/validators_test.go:82]: use `t.Run()` and omit the explicit log - you'll get them on failure as needed

File: internal/controllers/validators/validators_test.go [hunk 3]
```diff
@@ -1,63 +1,94 @@
...
+		operator := &v1alpha1.Operator{
+			Spec: v1alpha1.OperatorSpec{
+				Version: d.data,
+			},
+		}
+		if d.result {
+			require.NoError(t, validators.ValidateOperatorSpec(operator))
+		} else {
+			require.EqualError(t, validators.ValidateOperatorSpec(operator), fmt.Sprintf("invalid .spec.version: improper constraint: %s", d.data))
```
Comment (stevekuznetsov) [inline internal/controllers/validators/validators_test.go:91]: Seems like an overly brittle assertion on the specific format of the human-readable validation error - do we care about the specific contents here? Why? Will anyone ever programmatically look into this error message?

Comment (tmshort) [reply internal/controllers/validators/validators_test.go:91]: Can't answer the "why", but trying to keep the "what" as similar as possible. Although, AFAICT, all the errors are basically the same, so little point in checking the text.

File: internal/controllers/validators/validators_test.go (hunk 1 above)
Comment (tmshort) [reply internal/controllers/validators/validators_test.go:13]: This is literally a single-file test in this single-file package... I was having trouble making an anonymous struct, and gave up on it... I should go back and look.

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**